tests/automated_validation/test_comparison.py

```diff
@@ -230,34 +230,47 @@ def test_fuzzy_comparison_get_frame_aggregated(
     assert_frame_equal(diff, expected_df)
 
 
-def test_fuzzy_comparison_init_with_stratifications(
-    mock_ratio_measure: RatioMeasure,
-    test_data: dict[str, pd.DataFrame],
-    reference_data: pd.DataFrame,
-    reference_weights: pd.DataFrame,
-) -> None:
-    """Test that FuzzyComparison raises NotImplementedError when initialized with non-empty stratifications."""
-    with pytest.raises(
-        NotImplementedError, match="Non-default stratifications require rate aggregations"
-    ):
-        FuzzyComparison(
-            mock_ratio_measure,
-            DataSource.SIM,
-            test_data,
-            DataSource.GBD,
-            reference_data,
-            reference_weights,
-            stratifications=["year"],
-        )
-
-
-def test_fuzzy_comparison_get_frame_with_stratifications(
+@pytest.mark.parametrize("stratifications", [None, ["year"], []])
+@pytest.mark.parametrize("aggregate", [True, False])
+@pytest.mark.parametrize("draws", ["test", "reference", "both", "neither"])
+def test_fuzzy_comparison_get_frame_parametrized(
     mock_ratio_measure: RatioMeasure,
     test_data: dict[str, pd.DataFrame],
     reference_data: pd.DataFrame,
     reference_weights: pd.DataFrame,
+    stratifications: Collection[str] | None,
+    aggregate: bool,
+    draws: str,
 ) -> None:
     """Test that FuzzyComparison.get_frame raises NotImplementedError when called with non-empty stratifications."""
+    draw_values = list(
+        test_data["numerator_data"].index.get_level_values(DRAW_INDEX).unique()
+    )
+    if draws == "test":
+        # This is how the fixtures are set up
+        for key in test_data:
+            assert "input_draw" in test_data[key].index.names
+        assert "input_draw" not in reference_data.index.names
+        assert "input_draw" not in reference_weights.index.names
```

Comment:
`test_data` is a fixture, so aren't these all guaranteed to pass? Also, if we're making this check in the `"test"` case, shouldn't we also do it in the `"both"` case?

Reply:
I have removed these since they were not adding value to thetest.